mne_bids/tests/test_mne_bids.py

```diff
@@ -182,21 +188,41 @@ def test_kit():
     # ensure the channels file has no STI 014 channel:
     channels_tsv = make_bids_basename(
         subject=subject_id, session=session_id, task=task, run=run,
-        suffix='channels.tsv', acquisition=acq,
+        suffix='channels.tsv',
         prefix=op.join(output_path, 'sub-01/ses-01/meg'))
     data = _from_tsv(channels_tsv)
     assert 'STI 014' not in data['name']
 
     # ensure the marker file is produced in the right place
-    raw_folder = make_bids_basename(
-        subject=subject_id, session=session_id, task=task, run=run,
-        acquisition=acq, suffix='%s' % 'meg')
-    marker_fname = make_bids_basename(
-        subject=subject_id, session=session_id, task=task, run=run,
-        acquisition=acq, suffix='markers.sqd',
-        prefix=os.path.join(output_path, 'sub-01/ses-01/meg', raw_folder))
+    raw_folder = kit_bids_basename + '_meg'
+    marker_fname = os.path.join(output_path, 'sub-01/ses-01/meg', raw_folder,
+                                kit_bids_basename + '_markers.sqd')
     assert op.exists(marker_fname)
 
+    # test correct naming of marker files
+    raw = mne.io.read_raw_kit(
+        raw_fname, mrk=[hpi_post_fname, hpi_pre_fname], elp=electrode_fname,
```

Comment:
pre and post should change order?

Reply:
nope, I do this intentionally to confirm that even if they are provided in the wrong order it will correct it (as both have a measurement time)

Reply:
oh boy, too much magic ...

Reply:
I would invert the logic of the code a bit. I think it's more reasonable to say that the `pre` and `post` tags will always be based on the order in which they are provided. The date could be used as a validation instead. If the `pre` and `post` do not match the dates, then you raise an error

Reply:
but it is so nice making it all work like magic/easy :P

Reply:
I will invoke Zen of Python in my defense :P

```sh
$ python -c 'import this' | sed -n 14p
In the face of ambiguity, refuse the temptation to guess.
```

Reply:
By the way, I would argue that the `pre` and `post` is hard to disambiguate unless you also look at the start and stop time of the measurement. Probably not worth going into such a hassle in our code. You can simply distinguish them numerically based on the order they are provided in the hpi argument.

Reply:
Fair enough. I guess with my program I can do the "magic" to provide them in the correct order :P